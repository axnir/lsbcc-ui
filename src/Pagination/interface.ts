import { MouseEventHandler } from 'react';

export interface PaginationProps {
  /**
   * @description      数据总数
   */
  pageSize: number;
  /**
   * @description      每页条数
   */
  total?: number;
  /**
   * @description      当前页数
   */
  current: number;
  /**
   * @description      页码改变的回调，参数是改变后的页码
   */
  onChange(page: number): void;
}

export type OmitPageProps = {
  onClick: MouseEventHandler<HTMLElement>;
};
