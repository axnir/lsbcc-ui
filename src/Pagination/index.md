---
nav:
  title: 组件
  path: /components
  order: 2
group:
  title: 组件
---

# Pagination 分页器

分页器，采用分页的形式分隔长列表，每次只加载一个页面。

## 代码演示

### 基本用法

<code src="./demo/index.tsx"></code>

<API src="./pagination.tsx"></API>
