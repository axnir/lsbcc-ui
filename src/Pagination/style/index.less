@popupPrefix: lsbcc-pagination;

.@{popupPrefix} {
  .pages-inner {
    display: flex;
    flex-wrap: nowrap;
    list-style-type: none;
    padding: 0;

    li:hover {
      transform: scale(1.1);
      background-color: #f4f6f8;
    }

    .page {
      flex: 0 1 auto;
      height: 34px;
      line-height: 34px;
      cursor: pointer;
      min-width: 36px;
      min-height: 28px;
      display: inline-block;
      padding: 0 10px;
      margin: 0 4px;
      font-size: 14px;
      text-align: center;
      color: #606266;
      border: 1px solid #ebebeb;
      background-color: #fff;
      border-radius: 2px;
      box-sizing: border-box;

      span {
        user-select: none;
      }

      &.active {
        background-color: #4a96ff;
        color: #fff;
      }

      &.disabled {
        cursor: not-allowed;
      }
    }
  }
}
