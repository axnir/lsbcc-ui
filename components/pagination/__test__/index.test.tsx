import React, { useState } from 'react';
import { render } from '@testing-library/react';
import Pagination from '../pagination';

describe('<Pagination />', () => {
  const current = 1;

  test('should render default', () => {
    const { container } = render(
      <Pagination pageSize={10} onChange={console.log} current={current}>
        <button>left</button>
      </Pagination>,
    );
    expect(container).toMatchSnapshot();
  });
});
