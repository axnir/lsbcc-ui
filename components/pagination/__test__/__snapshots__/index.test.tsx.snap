// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render default 1`] = `
<div>
  <div
    class="lsbcc-pagination"
  >
    <ul
      class="pages-inner"
    >
      <li
        class="page btn disabled"
      >
        <span>
          &lt;
        </span>
      </li>
      <li
        class="page btn disabled"
      >
        <span>
          &gt;
        </span>
      </li>
    </ul>
  </div>
</div>
`;
